cluster/validation.go

```diff
@@ -674,6 +676,30 @@ func validatePodSecurityPolicy(c *Cluster) error {
 			return errors.New("PodSecurityPolicy has been removed and can not be enabled since k8s v1.25")
 		}
 	}
+	// check if there is any PSP resource when upgrading a cluster to k8s v1.25 and above
+	if parsedRangeAtLeast125(parsedVersion) {
+		kubeClient, err := k8s.NewClient(c.LocalKubeConfigPath, c.K8sWrapTransport)
+		if err != nil {
+			// we can not tell this is invoked when creating a new cluster or updating an existing one, so skip this check
+			logrus.Warnf("Skip the check for PSP resource due to the failire of initializing the kubernetes client")
+			return nil
+		}
+		pspList, _ := k8s.GetPSPList(kubeClient)
```

Comment:
Could pspList be nil? Why are we not returning a possible error? I'd suggest,

```
pspList, err := k8s.GetPSPList(kubeClient)
if err != nil {
    return errors.New("Error getting PodSecurityPolicy list for cluster")
}
if pspList == nil {
    return errors.New("PodSecurityPolicy list for cluster resource is invalid")
}
```

Reply:
I also thought about this and checked  [the source code](https://github.com/kubernetes/client-go/blob/v0.25.4/kubernetes/typed/policy/v1beta1/podsecuritypolicy.go#L86), it looks like a pointer is always returned even when nothing is found.